chips/earlgrey/src/pinmux.rs

```diff
@@ -0,0 +1,155 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! Mux'ing between physical pads and GPIO or other peripherals.
+
+use kernel::hil::gpio;
+use kernel::utilities::registers::interfaces::{Readable, Writeable};
+use kernel::utilities::registers::{register_bitfields, FieldValue, LocalRegisterCopy};
+use kernel::utilities::StaticRef;
+
+use crate::registers::pinmux_regs::{
+    PinmuxRegisters, DIO_PAD_ATTR_REGWEN, MIO_OUTSEL_REGWEN, MIO_PAD_ATTR_REGWEN,
+    MIO_PERIPH_INSEL_REGWEN,
+};
+use crate::registers::top_earlgrey::{
+    DirectPads, MuxedPads, PinmuxInsel, PinmuxOutsel, PinmuxPeripheralIn, PINMUX_AON_BASE_ADDR,
+};
+
+// To avoid code duplication for MIO/DIO we introcude
+// one register layout for bouth type of PAD. In feature this code
+// should be replaced by oficall improved auto generated definitions.
+// //TODO add links to gound true
+register_bitfields![u32,
+    pub(crate) PAD_ATTR [
+        INVERT OFFSET(0) NUMBITS(1) [],
+        VIRTUAL_OPEN_DRAIN_EN OFFSET(1) NUMBITS(1) [],
+        PULL_EN OFFSET(2) NUMBITS(1) [],
+        PULL OFFSET(3) NUMBITS(1) [
+            DOWN = 0,
+            UP = 1,
+        ],
+        KEEPER_EN OFFSET(4) NUMBITS(1) [],
+        SCHMITT_EN OFFSET(5) NUMBITS(1) [],
+        OPEN_DRAIN_EN OFFSET(6) NUMBITS(1) [],
+        SLEW_RATE OFFSET(16) NUMBITS(2) [],
+        DRIVE_STRENGTH OFFSET(20) NUMBITS(4) [],
+    ],
+];
+
+type PadAttribute = LocalRegisterCopy<u32, PAD_ATTR::Register>;
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum Pad {
+    Mio(MuxedPads),
+    Dio(DirectPads),
+}
+
+impl Pad {
+    // Extract value of attricttes using common layout
```

Comment:
s/"Extract value of attricttes using common layout"/"Extract the value of attributes using a common layout"